### DISCUSSION TITLE
Subscribing to an observable that is function of another observable

### DISCUSSION THREAD

**Discussion by martinrhan** [dotnet/reactive#1948] [Q&A]
I have an `IObservable<TypeA>` instance. There is a function `public static IObservable<TypeB> MyFunction(TypeA a)`.Now I want to subscribe to the `IObservable<TypeB>` object returned from `MyFunction(currentTypeAObject)`, but when the `IObservable<TypeA>` has a change, the previous subscription of the `TypeB` object is disposed and a new subscription to whats returned by `MyFunction(newTypeAObject)` is established.

Whats decribed above is actually a generalization of a common scenario in UI development, that is, subscribing to a nested property, for example, given `objectA`, subscribe `objectA.TypeAPropertyA.TypeBPropertyB`. Existing frameworks somehow provide capability to subscribe the entire path, in which any part of the chain changes result into a notify to the subsribers and internal modifications of subscription states. However, they are limited and not generalized. For instance, ReactiveUI only support binding path that all parts are properties. 

 I would like to know if there is an elegant and swift solution for this, using existing classes in reactive extensions.

**Comment by fedeAlterio** [comment]
Exists an operator that does exactly what you ask: Switch.
`observableA.Select(MyFunction).Switch()`
Returns an `IObservale<B>` that produces values from the last observable returned from `MyFunction`. Basically every time a new element of a is produced, it unsubscribes from the previous `IObservale<B>` and subscribes to the new one.

`this.WhenAnyValue(x => x.PropA)
.Select(y => y.WhenAnyValue(z => z.PropB).Switch()` you should be careful with nulls, but this is the idea
